### PR TITLE
add scenario test cases

### PR DESCRIPTION
**Pull request by MikeStall**
Add scenario test cases demonstrating various usages on IUntypedObject.
Fix bug hit by test.

### CODE REVIEW

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/ScenarioDotNetObjectWrapper.cs [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        private static void Add(RecalcEngine engine, string name, object obj)
+        {
+            var objFx = FormulaValue.New(new Wrapper(obj));
+            engine.UpdateVariable(name, objFx);
+        }
+
+        [Theory]        
+        [InlineData("Value(obj.Next.Value)", 20.0)]
+        [InlineData("obj.missing", null)]
+        [InlineData("IsBlank(obj.Next.Next)", true)]
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/ScenarioDotNetObjectWrapper.cs:40]: >[InlineData("IsBlank(obj.Next.Next)", true)] [](http://example.com/codeflow?start=8&length=44)

add a false case

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/ScenarioDotNetObjectWrapper.cs [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+            engine.UpdateVariable(name, objFx);
+        }
+
+        [Theory]        
+        [InlineData("Value(obj.Next.Value)", 20.0)]
+        [InlineData("obj.missing", null)]        
+        [InlineData("IsBlank(obj.Next.Next)", true)]
+        [InlineData("IsBlank(obj.Next)", false)]
+        [InlineData("obj.Next.Next", null)]
+        [InlineData("IsBlank(Index(array, 3))", true)]
```
Comment (Shpakh-MSFT) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/ScenarioDotNetObjectWrapper.cs:44]: >ta("IsBlank(Index(array, 3))" [](http://example.com/codeflow?start=17&length=29)

You may want to add tests for out of bound index if not already present.

**Review by Shpakh-MSFT — APPROVED** [review]
:shipit:

**Comment by MikeStall** [comment]
>                 return new BlankValue(IRContext.NotInSource(FormulaType.Blank));

For out-of-bounds, Excel returns error  (not blank) 
@bryancroteau-MSFT 

---
Refers to: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs:40 in 63e1853. [](commit_id = 63e1853131cb9b27972832e2208068b07420d32e, deletion_comment = False)

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/ScenarioDotNetObjectWrapper.cs (hunk 2 above)
Comment (MikeStall) [reply src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/ScenarioDotNetObjectWrapper.cs:44]: added. 
Current behavior is a bit odd: too low is an error, too high is blank.

Comment (Shpakh-MSFT) [reply src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/ScenarioDotNetObjectWrapper.cs:44]: Agree that both should be errors.

**Comment by MikeStall** [comment]
>                 return new BlankValue(IRContext.NotInSource(FormulaType.Blank));

updated to be consistent with excel. 

---
In reply to: [1048048838](https://github.com/microsoft/Power-Fx/pull/172#issuecomment-1048048838) [](http://example.com/codeflow?ancestors=1048048838,1048048838)

---
Refers to: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs:40 in 63e1853. [](commit_id = 63e1853131cb9b27972832e2208068b07420d32e, deletion_comment = False)

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/ScenarioDotNetObjectWrapper.cs [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+            }
+
+            public bool TryGetProperty(string value, out IUntypedObject result)
+            {
+                var t = _source.GetType();
+                var prop = t.GetProperty(value, BindingFlags.Public | BindingFlags.Instance);
+                if (prop == null)
+                {
+                    result = null;
+                    return false;
```
[the remark this answers is not in the document]
Comment (MikeStall) [reply src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/ScenarioDotNetObjectWrapper.cs:218]: discussed - fx semantics are to return blanks.

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/ScenarioDotNetObjectWrapper.cs [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+            Assert.True(ReferenceEquals(obj1result, obj1));
+        }
+
+        [Fact]
+        public void ObjectTypes()
+        {
+            // FormulaType.UntypedObject is any value wrapping a IUntypedObject
+            // IUntypedObject can represent any Fx type as well as foriegn types. 
+            // ExternalType.ObjectType is the set of IUntypedObject that represent a foriegn object 
+            Assert.NotEqual(FormulaType.UntypedObject, ExternalType.ObjectType);
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/ScenarioDotNetObjectWrapper.cs:108]: >FormulaType.UntypedObject, ExternalType.ObjectType [](http://example.com/codeflow?start=28&length=50)

These are easy to confuse (I hit made a few mistakes consuming them), so I added a test case and comment to be explicit.

**Review by lesaltzm — APPROVED** [review]

**Pull request merged**